Tests/MapboxMapsTests/MapView/Integration Tests/OfflineManagerIntegrationTests.swift

```diff
@@ -0,0 +1,158 @@
+            DispatchQueue.main.async {
+                downloadInProgress.fulfill()
+                downloadInProgress.assertForOverFulfill = false
+            }
+        } completion: { result in
+            switch result {
+            case .success(let region):
+                completionBlockReached.fulfill()
+            case .failure(let error):
+                XCTFail("Download failed with error: \(error)")
+            }
+        }
+
+        let expectations = [downloadInProgress, completionBlockReached]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testProgressCanBeCancelled() {
+
+        /// Expectations to be fulfilled
+        let downloadInProgress = XCTestExpectation(description: "Downloading offline tiles in progress")
+        let downloadWasCancelled = XCTestExpectation(description: "Checks a cancel function was reached and that the download was canceled")
+
+        /// Perform the download
+        let download = TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                                              loadOptions: tileRegionLoadOptions!) { _ in
+            DispatchQueue.main.async {
+                downloadInProgress.fulfill()
+            }
+        } completion: { _ in }
+
+        /// This guarantees that after 3 seconds of a download in progress, we will force a cancel
+        DispatchQueue.main.asyncAfter(deadline: .now() + 3.0) {
+            self.cancelDownload(download: download, expectation: downloadWasCancelled)
+        }
+
+        let expectations = [downloadInProgress, downloadWasCancelled]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testOfflineRegionCanBeDeleted() {
+
+        /// Expectations to be fulfilled
+        let downloadWasDeleted = XCTestExpectation(description: "Downloaded offline tiles were deleted")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                               loadOptions: tileRegionLoadOptions!) { _ in } completion: { _ in }
+
+        TileStore.getInstance().removeTileRegion(forId: self.tileRegionId)
+
+        TileStore.getInstance().allTileRegions(completion: { result in
+            switch result {
+            case .success(let tileRegions):
+                if tileRegions.count == 0 {
+                    downloadWasDeleted.fulfill()
+                }
+            case .failure(let error):
+                XCTFail("Error getting tile regions with error: \(error)")
+            }
+        })
+
+        let expectations = [downloadWasDeleted]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testMapCanBeLoadedWithNetworkConnectivity() {
+        /// Expectations to be fulfilled
+        let mapDidLoad = XCTestExpectation(description: "Map was loaded")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
```

Comment:
Does this depend on a previous test - if so it might have been deleted.

Rather than using `.mapLoaded` we should use `.resourceRequest` and check the event's [`data-source`](https://github.com/mapbox/mapbox-maps-ios/blob/699cdd28fcef21bcbdd21831275e1692f703ba13/Sources/MapboxMaps/Foundation/Extensions/Core/MapEvents.swift#L221):

Reply:
okay I can do that